torch/distributed/_shard/sharding_spec/api.py

```diff
@@ -92,174 +88,60 @@ def shard(self, tensor: torch.Tensor, src_rank: int = 0, process_group=None) ->
             A :class:`ShardedTensor` sharded from the given tensor.
         """
 
-@dataclass
-class ChunkShardingSpec(ShardingSpec):
-    """
-    This is a type of PlacementSpec that defines the placement as being sharded
-    across multiple devices. In particular, it represents sharding a Tensor
-    along a single dimension into equal chunks (similar to :meth:`torch.chunk`).
+    @classmethod
+    def _register_custom_op(cls, sharding_spec: Type, op: Callable, func: Callable):
+        """
+        Allows registration of a custom op for ShardedTensor to enable
+        custom optimizations for a particular ShardingSpec.
+        Args:
+            sharding_spec(type): The ShardingSpec for which we need to add this custom op.
+            op(Callable): The op to override (ex: torch.bmm)
+            func(Callable): The custom implementation for ``op``
+        """
+        from inspect import signature
+        if len(signature(func).parameters) != 3:
+            raise TypeError(
+                f'Custom sharded op function expects signature: '
+                f'(types, args, kwargs), but received '
```

Comment:
for custom_spec_op, we don't pass process_group, are we assuming that we would use the sharded tensor associated process group for communication in the op implementation? what if two parameters in the module have different process group, should we check that they are in the same pg before doing the computation?